crates/bevy_ecs/src/system/query.rs

```diff
@@ -50,330 +57,424 @@ use core::{
+/// if an entity has both the `Player` and `Enemy` components. Bevy will catch this and panic,
+/// however, instead of breaking Rust's mutability rules:
 ///
 /// ```should_panic
 /// # use bevy_ecs::prelude::*;
+/// #
 /// # #[derive(Component)]
 /// # struct Health;
+/// #
 /// # #[derive(Component)]
 /// # struct Player;
+/// #
 /// # #[derive(Component)]
 /// # struct Enemy;
 /// #
 /// fn randomize_health(
 ///     player_query: Query<&mut Health, With<Player>>,
 ///     enemy_query: Query<&mut Health, With<Enemy>>,
-/// )
-/// # {}
-/// # let mut randomize_health_system = IntoSystem::into_system(randomize_health);
-/// # let mut world = World::new();
-/// # randomize_health_system.initialize(&mut world);
-/// # randomize_health_system.run((), &mut world);
+/// ) {
+///     // ...
+/// }
+/// #
+/// # bevy_ecs::system::assert_system_does_not_conflict(randomize_health);
 /// ```
 ///
-/// Adding a `Without` filter will disjoint the queries.
-/// In this way, any entity that has both `Player` and `Enemy` components is excluded from both queries.
+/// Adding a [`Without`] filter will disjoint the queries. In the following example, any entity
+/// that has both the `Player` and `Enemy` components will be excluded from _both_ queries:
 ///
 /// ```
 /// # use bevy_ecs::prelude::*;
+/// #
 /// # #[derive(Component)]
 /// # struct Health;
+/// #
 /// # #[derive(Component)]
 /// # struct Player;
+/// #
 /// # #[derive(Component)]
 /// # struct Enemy;
 /// #
 /// fn randomize_health(
 ///     player_query: Query<&mut Health, (With<Player>, Without<Enemy>)>,
 ///     enemy_query: Query<&mut Health, (With<Enemy>, Without<Player>)>,
-/// )
-/// # {}
-/// # let mut randomize_health_system = IntoSystem::into_system(randomize_health);
-/// # let mut world = World::new();
-/// # randomize_health_system.initialize(&mut world);
-/// # randomize_health_system.run((), &mut world);
+/// ) {
+///     // ...
+/// }
+/// #
+/// # bevy_ecs::system::assert_system_does_not_conflict(randomize_health);
 /// ```
 ///
-/// An alternative to this idiom is to wrap the conflicting queries into a [`ParamSet`](super::ParamSet).
+/// An alternative solution to this problem would be to wrap the conflicting queries in
+/// [`ParamSet`].
+///
+/// [`Without`]: crate::query::Without
+/// [`ParamSet`]: crate::system::ParamSet
 ///
 /// ## Whole Entity Access
 ///
-/// [`EntityRef`]s can be fetched from a query. This will give read-only access to any component on the entity,
-/// and can be used to dynamically fetch any component without baking it into the query type. Due to this global
-/// access to the entity, this will block any other system from parallelizing with it. As such these queries
-/// should be sparingly used.
+/// [`EntityRef`] can be used in a query to gain read-only access to all components of an entity.
+/// This is useful when dynamically fetching components instead of baking them into the query type.
```

Comment:
Is this just not true anymore? 

`Due to this global access to the entity, this will block any other system from parallelizing with it. As such these queries should be sparingly used.`

Reply:
I added this back to the Performance section!